### ISSUE TITLE
"Connection failed" when connecting to nas with SMB2 - no apparent activity

### ISSUE THREAD

**Issue by whitty** [wa2c/cifs-documents-provider#10]
This may be a duplicate of #8, but I didn't want to hijack it.

I originally had SMB1 for my NAS - but was able to update it via a plug in.

I can browse the shares as anonymous (to avoid possible password mishaps)

```
smbclient -L //192.168.1.15 -U "%"
	Sharename       Type      Comment
	---------       ----      -------
	IPC$            IPC       IPC Service (nas)
	video           Disk      
	photos          Disk      
	music           Disk            
SMB1 disabled -- no workgroup available
```

I can see on the nas's SMB logs that activity is occurring.  If I do password authentication I also see that in the log (eg `NT_STATUS_WRONG_PASSWORD` or `succeeded`).  So I'm convinced the NAS is responding OK.

However when I configure a connection in CIFS Documents Provider:
* using IP address for host
* using anonymous (also using username)
* with or without "folder" selected. 
* (all the rest empty or default)
I get the toast notification `Connection failed`.

There's no indication of SMBD activity in the nas's SMB logs, so it feels like no connection was attempted.

I tried to download the APK you referenced, but it returns a 404.  

Do I understand #8 correctly - is the current version of CIFS provided refusing to talk to SMB2 devices?  Is that what the extra APK was?

**Comment by wa2c** [comment]
@whitty 
I released 1.4.0 beta test version on Google Play.
I think this version solves the problem.
Can you try this and see if it has been fixed?

**Comment by whitty** [comment]
No change in behaviour - is there somewhere I can get debug info to send?

**Comment by wa2c** [comment]
@whitty
I released 1.4.0 new beta test version on Google Play.
It is improved some problems.

**Comment by whitty** [comment]
Thanks - I installed the latest update

1. thanks for adding the error code to the toast notification - that helped a lot - I could check connection by doing anonymous and showing I got an auth error, before 
2. I was now able to complete the connection and can use the connection in other app.

Looks good ;)
